python/test/backend/test_device_backend.py

```diff
@@ -43,9 +44,10 @@ def build_for_backend(name, src, srcdir):
         scheme = 'posix_prefix'
     py_include_dir = sysconfig.get_paths(scheme=scheme)["include"]
 
-    ret = subprocess.check_call([cc, src, f"-I{py_include_dir}", f"-I{srcdir}", "-shared", "-fPIC", "-o", so])
-    if ret == 0:
-        return so
+    with contextlib.suppress(subprocess.CalledProcessError):
+        ret = subprocess.check_call([cc, src, f"-I{py_include_dir}", f"-I{srcdir}", "-shared", "-fPIC", "-o", so])
```

Comment:
Here are two ways of writing this that I think would be more Pythonic.

1. Use run() instead of check_call() so you don't have to suppress the exception.
2. Use check_call but use a try-catch.

```
try:
  check_call(...)
  return
except CalledProcessError:
  # fallback to setuptools below
  pass
```

Reply:
Isn't (2) equivalent? That pattern was what `suppress` was created for in order to make more readable/clearer intent.

I'll rework it for with (1).